src/rhsmlib/services/register.py

```diff
@@ -132,6 +132,16 @@ def register(
             cp_provider = inj.require(inj.CP_PROVIDER)
             cp_provider.close_all_connections()
 
+        access_mode: str = consumer.get("owner", {}).get("contentAccessMode", "unknown")
+        if access_mode != "org_environment":
+            log.error(
+                f"Organization's content access mode is '{access_mode}'. "
+                "Only simple content access/'org_environment' is allowed."
+            )
+            raise exceptions.ServiceError(
+                _("Registration is only possible when the organization is in Simple Content Access Mode.")
```

Comment:
a minor wording note, according to what I see in other user-facing messages: s/Simple Content Access Mode/Simple Content Access (SCA) mode/
- lowercase "mode" (it is not part of the acronym
- add "(SCA)" as acronym, so it can be spotted even by those that don't know the "expanded name"